File: 05_Open_source_ProjectRepository/AIAgentFramework/EnlightenmentLighthouseRuntime/examples/python/hello.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
Hello World example for Enlightenment Lighthouse Runtime (ELR)
This example demonstrates how to run a Python application in ELR
"""

import os
import sys
import time

# Print welcome message
print("====================================")
print("Hello from Enlightenment Lighthouse Runtime!")
print("====================================")
print(f"Language: Python {sys.version}")
print(f"Runtime: ELR")
print("====================================")

# Demonstrate basic Python features
print("\nBasic Python Features:")

# Variables and output
number = 42
message = "The answer to life, the universe, and everything"
print(f"Integer variable: {number}")
print(f"String variable: {message}")

# Loop
print("\nLoop demonstration:")
for i in range(1, 6):
    print(f"Iteration {i}")

# List
print("\nList demonstration:")
languages = ["C++", "Python", "Java", "JavaScript", "Go"]
for lang in languages:
    print(f"Supported language: {lang}")

# Function
print("\nFunction demonstration:")

def square(x):
    """Calculate square of a number"""
    return x * x

for i in range(1, 4):
    print(f"Square of {i} is {square(i)}")

# Environment variables
print("\nEnvironment variables:")
elr_container_id = os.environ.get("ELR_CONTAINER_ID", "Not set (running outside ELR)")
print(f"ELR_CONTAINER_ID: {elr_container_id}")

# Try to import common libraries
print("\nLibrary demonstration:")

try:
    import numpy as np
    print("NumPy: Available")
    # Simple NumPy example
    arr = np.array([1, 2, 3, 4, 5])
    print(f"NumPy array: {arr}")
    print(f"Array mean: {np.mean(arr)}")
except ImportError:
    print("NumPy: Not available")

try:
    import requests
    print("Requests: Available")
except ImportError:
    print("Requests: Not available")

# End message
print("\n====================================")
print("Python example completed successfully!")
print("====================================")

File: 05_Open_source_ProjectRepository/AIAgentFramework/EnlightenmentLighthouseRuntime/examples/cpp/hello.cpp
// Hello World example for Enlightenment Lighthouse Runtime (ELR)
// This example demonstrates how to run a C++ application in ELR

#include <iostream>
#include <string>
#include <vector>

int main() {
    // Print welcome message
    std::cout << "====================================" << std::endl;
    std::cout << "Hello from Enlightenment Lighthouse Runtime!" << std::endl;
    std::cout << "====================================" << std::endl;
    std::cout << "Language: C++" << std::endl;
    std::cout << "Runtime: ELR" << std::endl;
    std::cout << "====================================" << std::endl;
    
    // Demonstrate basic C++ features
    std::cout << "\nBasic C++ Features:" << std::endl;
    
    // Variables and output
    int number = 42;
    std::string message = "The answer to life, the universe, and everything";
    std::cout << "Integer variable: " << number << std::endl;
    std::cout << "String variable: " << message << std::endl;
    
    // Loop
    std::cout << "\nLoop demonstration:" << std::endl;
    for (int i = 1; i <= 5; ++i) {
        std::cout << "Iteration " << i << std::endl;
    }
    
    // Vector
    std::cout << "\nVector demonstration:" << std::endl;
    std::vector<std::string> languages = {"C++", "Python", "Java", "JavaScript", "Go"};
    for (const auto& lang : languages) {
        std::cout << "Supported language: " << lang << std::endl;
    }
    
    // Function
    std::cout << "\nFunction demonstration:" << std::endl;
    
    // Lambda function to calculate square
    auto square = [](int x) {
        return x * x;
    };
    
    for (int i = 1; i <= 3; ++i) {
        std::cout << "Square of " << i << " is " << square(i) << std::endl;
    }
    
    // Environment variables
    std::cout << "\nEnvironment variables:" << std::endl;
    if (const char* env_p = std::getenv("ELR_CONTAINER_ID")) {
        std::cout << "ELR_CONTAINER_ID: " << env_p << std::endl;
    } else {
        std::cout << "ELR_CONTAINER_ID: Not set (running outside ELR)" << std::endl;
    }
    
    // End message
    std::cout << "\n====================================" << std::endl;
    std::cout << "C++ example completed successfully!" << std::endl;
    std::cout << "====================================" << std::endl;
    
    return 0;
}
